### ISSUE TITLE
Improve approximate/fuzzy string matching in search

### ISSUE THREAD

**Issue by GuyUnger** [godotengine/godot-proposals#7771]
### Describe the project you are working on

NA

### Describe the problem or limitation you are having in your project

It's hard to find files because Godot requires very exactly queries.
In the file search the only improvement is that you can search in the middle of names
Quick open allows for missing letters but it's not exactly helpful, for example here I misspell "camera" by one letter and the only result is completely unrelated:
![Godot_4_JEXPR5OIIp](https://github.com/godotengine/godot-proposals/assets/7023847/b7725127-4c32-4ffa-95c1-9c5475dc0790)

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Two things that would improve this:
- Instead of including spaces as part of the search term, make them split the query into multiple terms. 
- Allow room for typos, such as with the leuvhenstein distance

I think this would mainly be helpful for the file search and quick open. For filters for the tree, properties, and settings etc. it might not be helpful because items don't get sorted by relevance so could easily clutter the results with things that you didn't want to search for.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Examples of searches and what they should find:

"big enemy" -> "enemy_big_hit.wav"
"plater" -> "player.tscn"

### If this enhancement will not be used often, can it be worked around with a few lines of script?

NA

### Is there a reason why this should be core and not an add-on in the asset library?

NA

**Comment by Calinou** [comment]
There have been many pull requests experimenting with fuzzy search in the search help dialog (for instance), but it often turned out to result in usability regressions in practice. It's hard to have an algorithm that *always* provides a straight usability upgrade, even in difficult situations (such as searches with very short strings).

**Comment by samsface** [comment]
@Calinou Does Godot have any concept of flights/feature flags? So, we can try put this under a feature flag and if in practice it sucks, we can remove it?

**Comment by Calinou** [comment]
> Does Godot have any concept of flights/feature flags?

No, other than creating pre-release builds with an editor setting that can be disabled.

**Comment by samsface** [comment]
Ok cool, I'll open a PR for this soon 🔜

**Comment by KoBeWi** [comment]
Implemented by https://github.com/godotengine/godot/pull/98278
